examples/tree-view/src/Node/treeSlice.js

```diff
@@ -0,0 +1,67 @@
+import { createSlice } from "@reduxjs/toolkit";
+import generateTree from "../generateTree";
+
+const { tree, idCount } = generateTree();
+
+let nextId = idCount;
+const initialState = tree;
+
+const getAllDescendants = (tree, id) => {
+  const descendants = [id];
+  const run = (id) => {
+    tree[id].childIds.forEach((childId) => {
+      descendants.push(childId);
+      run(childId);
+    });
+  };
+
+  run(id);
+  return descendants;
+};
+
+const deleteMany = (tree, ids) => {
+  ids.forEach((id) => delete tree[id]);
+};
+
+const treeSlice = createSlice({
+  name: "tree",
+  initialState,
+  reducers: {
+    increment(state, action) {
+      const nodeId = action.payload;
+      state[nodeId].counter++;
+    },
+    createNode(state) {
+      nextId++;
+      state[nextId] = { id: nextId, counter: 0, childIds: [] };
+    },
+    deleteNode(state, action) {
+      const nodeId = action.payload;
+      const ids = getAllDescendants(state, nodeId);
+
+      deleteMany(state, ids);
+    },
+    addChild(state, action) {
+      const nodeId = action.payload;
+      const childId = nextId;
+      state[nodeId].childIds.push(childId);
```

Comment:
:speech_balloon: I haven't even looked at the old code, but I guarantee this is _wayyyy _better than the hand-written immutable update version :)